yandex_contest/binary_sum.py

```diff
@@ -1,44 +1,35 @@
-def convert(string):
-    list_1 = []
-    list_1[:0] = string
-    list_x = [int(i) for i in list_1]
+def convert_to_list(string):
+    list_x = [int(i) for i in string]
     return list_x
 
+
 def main():
     num_1 = input()
-    list_1 = convert(num_1)
+    list_1 = convert_to_list(num_1)
 
     num_2 = input()
-    list_2 = convert(num_2)
+    list_2 = convert_to_list(num_2)
+
+    reversed_num_1 = list(reversed(list_1))
+    reversed_num_2 = list(reversed(list_2))
+
+    if len(reversed_num_1) > len(reversed_num_2):
+        temp = reversed_num_1
+        reversed_num_1 = reversed_num_2
+        reversed_num_2 = temp
+
+    for _ in range(len(reversed_num_2) - len(reversed_num_1)):
+        reversed_num_1.append(0)
 
     result = []
     summa = 0
     rest = 0
 
-    number1 = list(reversed(list_1))
-    number2 = list(reversed(list_2))
-
-    diff = len(list_1) - len(list_2)
-    if diff < 0:
-        for _ in range(abs(diff)):
-            number1.append(0)
-    elif diff > 0:
-        for _ in range(abs(diff)):
-            number2.append(0)
-
-    for i in range(len(number1)):
-        summa = number1[i] + number2[i] + rest
-        if summa == 2:
-            summa = 0
-            rest = 1
-            result.append(summa)
-        elif summa == 3:
-            summa = 1
-            rest = 1
-            result.append(summa)
-        else:
-            result.append(summa)
-            rest = 0
+    for i in range(len(reversed_num_1)):
+        summa = reversed_num_1[i] + reversed_num_2[i] + rest
+        rest = summa // 2
+        summa = summa % 2
```

Comment:
Great stuff! We can also shorten this a bit
```suggestion
        summa %= 2
```